src/domain/Player.java

```diff
@@ -18,4 +19,30 @@ public String getName() {
 	public void giveCard(Card c) {
 		cards.add(c);
 	}
+	
+	//Give players a row of cards
+	public void giveCardRow(List<Card> cards) {
+		
+	}
+	
+	//Calculate score per player
+	public int calculateScore() {
+		int score = 0;
+		int aantal = 0;
+		Color color;
+		
+		while(!cards.isEmpty()) {
+			color = cards.get(1).getColor();
+			cards.remove(0);
+			aantal++;
+			for(Card card : cards) {
+				if(card.getColor() == color) {
+					cards.remove(card);
+					aantal++;
+				}
+			}
+			score += aantal;
+		}
+		return score;
```

Comment:
Deze methode is goed, maar je kan volgens mij de `score` variabele weg halen en gewoon `aantal` teruggeven. Want op het einde zijn `score` en `aantal` toch hetzelfde.

Reply:
Ik denk niet dat score en aantal uiteindelijk hetzelfde is. Voor de eerste kaart wel, maar als je een tweede kaart van dezelfde kleur hebt krijg je drie punten, voor drie kaarten van dezelfde kleur zes punten,... Dit gaat door tot 6 of meer van dezelfde kleur, daarvoor krijg je 21 punten.

Reply:
Je oplossing was juist, maar door het gebruik van een enhanced for loop krijg je een `ConcurrentModificationException`, dit kan opgelost worden door deze code te gebruiken:

```java
	public int calculateScore() {
		int score = 0;
		int aantal = 0;
		Color color;

		while(!cards.isEmpty()) {
			color = cards.get(1).getColor();
			cards.remove(0);
			aantal++;

			Iterator<Card> cardIterator = cards.iterator();
			while (cardIterator.hasNext()) {
				Card card = cardIterator.next();

				if(card.getColor() == color) {
					cardIterator.remove();
					aantal++;
				}
			}

			score += aantal;
			aantal = 0;
		}
		return score;
	}
```

Ik zal misschien een paar JUnit testen schrijven zodat we dit soort dingen gemakkelijker kunnen testen.

Reply:
Ah, okido, thanks. Als er nog iets aangepast moet worden, zeg maar en ik zal proberen wat ik kan doen.